### ISSUE TITLE
Error in v3.4: Couldn't find file 'locomotive/vendor'

### ISSUE THREAD

**Issue by JacobEvelyn** [locomotivecms/engine#1262]
#1244 was closed, but I'm still running into issues trying to get Locomotive 3.4 working.

With a relatively fresh install (I ran `bundle exec rails generate locomotive:install`, though I might've made other changes a month or two ago when I was last working on this upgrade that I don't remember), I'm getting this error:

```
ActionView::Template::Error (couldn't find file 'locomotive/vendor' with type 'application/javascript'
Checked in these paths: 
  /panorama/locomotiveapp/app/assets/config
  /panorama/locomotiveapp/app/assets/images
  /panorama/locomotiveapp/app/assets/javascripts
  /panorama/locomotiveapp/app/assets/stylesheets
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/locomotivecms-3.4.0/app/assets/images
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/locomotivecms-3.4.0/app/assets/javascripts
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/locomotivecms-3.4.0/app/assets/stylesheets
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/jquery-ui-rails-6.0.1/app/assets/images
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/jquery-ui-rails-6.0.1/app/assets/javascripts
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/jquery-ui-rails-6.0.1/app/assets/stylesheets
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/jquery-rails-4.3.3/vendor/assets/javascripts
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/codemirror-rails-5.16.0/vendor/assets/javascripts
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/codemirror-rails-5.16.0/vendor/assets/stylesheets
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/actioncable-5.1.6/lib/assets/compiled
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/actionview-5.1.6/lib/assets/compiled
  /panorama/locomotiveapp/node_modules
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/bootstrap-sass-3.3.7/assets/stylesheets
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/bootstrap-sass-3.3.7/assets/javascripts
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/bootstrap-sass-3.3.7/assets/fonts
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/bootstrap-sass-3.3.7/assets/images
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/font-awesome-sass-5.0.13/assets/stylesheets
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/font-awesome-sass-5.0.13/assets/fonts
  /Users/jacobevelyn/.gem/ruby/2.5.1/gems/locomotivecms-3.4.0/vendor/assets/components/locomotive):
    1: - title t('locomotive.registrations.new.title')
    2: 
    3: = link_to root_path, class: 'brand-logo' do
    4:   = account_logo_tag
    5: 
    6: .public-box
    7:   .public-intro
  
locomotivecms (3.4.0) app/assets/javascripts/locomotive.js:13
...
```

Is there some setup step I'm missing?

**Comment by JacobEvelyn** [comment]
Ping @did

**Comment by did** [comment]
hi @JacobEvelyn! No pb, I'll have @gridar take a look at it tomorrow!

**Comment by JacobEvelyn** [comment]
Just a friendly ping, @gridar 😄

**Comment by gridar** [comment]
Hi @JacobEvelyn, can you check the folders you have at the root of your project ?
It should contain a "vendor" folder with a .keep file to let Git knows it's important.

**Comment by JacobEvelyn** [comment]
Hi @gridar—yes, there is a `vendor/.keep` file in my project.

If it's helpful, this is from a pretty basic setup—I created a new Rails app following the Rails setup guides, then changed the `Gemfile` to only include:

```
gem "locomotivecms", "~> 3.4"
gem "carrierwave-aws"
```

and then ran `rails generate locomotive:install` and configured the Locomotive settings. I unfortunately don't remember more details of what I've done because I started this upgrade process (to Locomotive 3.4) months ago and have had a hard time getting prompt answers here (see: #1244) when things go wrong. 🙁

Any help on this would be much appreciated!

**Comment by gridar** [comment]
Alright I reproduced your bug and found a strange solution: 
try to replace your 
``` gem "locomotivecms", "~> 3.4" ```
with
``` gem 'locomotivecms', github: 'locomotivecms/engine', ref: '9ed6083' ```
it's the same commit but somehow it trigger the bug you mention on my fresh locomotive install ....
Can you tell me if it worked ?

**Comment by gridar** [comment]
@JacobEvelyn Any news on your side about this ?

**Comment by JacobEvelyn** [comment]
Thanks for the help, @gridar! Looks like that fixed the issue. I wonder if the version pushed to RubyGems was corrupted in some way, or was accidentally pushed as a different version of the code? (Or you might want to confirm that it wasn't a malicious push from an attacker.)

**Comment by gridar** [comment]
Seems the tag and release mismatch, we'll fix it along with the doc quickly, Thanks.

**Comment by gnoireaux** [comment]
got the same issue following: https://doc.locomotivecms.com/docs/getting-started-with-locomotive

**Comment by gnoireaux** [comment]
`gem 'locomotivecms', github: 'locomotivecms/engine', ref: '9ed6083'`
solved the issue for me too.

Thanks for the solution @gridar !  Wanted to ping you also because if you thought you fixed the issue, then here's the info telling you it is not. If you have not yet re-released, then everything's ok :)

So I changed the Gemfile and then `bundle update`.  The signup page got rendered without error but it took a long time. If it is useful to you, see at the bottom of the comment: first rendering took almost a minute.

```
Fetching https://github.com/locomotivecms/engine.git
The dependency tzinfo-data (>= 0) will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
Fetching gem metadata from https://rubygems.org/.........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...........
Using rake 12.3.1
Using RedCloth 4.3.2
Using concurrent-ruby 1.1.3
Using i18n 1.1.1
Using minitest 5.11.3
Using thread_safe 0.3.6
Using tzinfo 1.2.5
Using activesupport 5.1.6.1
Using builder 3.2.3
Using erubi 1.7.1
Using mini_portile2 2.3.0
Using nokogiri 1.8.5
Using rails-dom-testing 2.0.3
Using crass 1.0.4
Using loofah 2.2.3
Using rails-html-sanitizer 1.0.4
Using actionview 5.1.6.1
Using rack 2.0.6
Using rack-test 1.1.0
Using actionpack 5.1.6.1
Using nio4r 2.3.1
Using websocket-extensions 0.1.3
Using websocket-driver 0.6.5
Using actioncable 5.1.6.1
Using globalid 0.4.1
Using activejob 5.1.6.1
Using mini_mime 1.0.1
Using mail 2.7.1
Using actionmailer 5.1.6.1
Using activemodel 5.1.6.1
Using arel 8.0.0
Using activerecord 5.1.6.1
Using bundler 1.17.1
Using method_source 0.9.2
Using thor 0.20.3
Using railties 5.1.6.1
Using sprockets 3.7.2
Using sprockets-rails 3.2.1
Using rails 5.1.6.1
Using actionmailer-with-request 0.5.0
Using public_suffix 3.0.3
Using addressable 2.5.2
Using attr_extras 5.2.0
Using execjs 2.7.0
Using autoprefixer-rails 8.0.0
Using descendants_tracker 0.0.4
Using ice_nine 0.11.2
Using axiom-types 0.1.1
Using bazaar 0.0.2
Using bcrypt 3.1.12
Using bindex 0.5.0
Using kaminari-core 1.1.1
Using kaminari-actionview 1.1.1
Using kaminari-activerecord 1.1.1
Using kaminari 1.1.1
Using bootstrap-kaminari-views 0.0.5
Using sass 3.4.25
Using bootstrap-sass 3.3.7
Using bson 4.3.0
Using byebug 10.0.2
Using xpath 3.2.0
Using capybara 2.18.0
Using mime-types-data 3.2018.0812
Using mime-types 3.1
Using carrierwave 1.2.3
Using mongo 2.6.2
Using mongoid 6.4.2
Using mongoid-grid_fs 2.4.0
Using carrierwave-mongoid 1.0.0
Using ffi 1.9.25
Using childprocess 0.9.0
Using chronic 0.10.2
Using chunky_png 1.3.11
Using codemirror-rails 5.16.0
Using coercible 1.0.0
Using coffee-script-source 1.12.2
Using coffee-script 2.4.1
Using coffee-rails 4.2.2
Using colorize 0.8.1
Using multi_json 1.13.1
Using compass-core 1.0.3
Using compass-import-once 1.0.5
Using rb-fsevent 0.10.3
Using rb-inotify 0.9.10
Using compass 1.0.3
Using money 6.13.1
Using monetize 1.7.0
Using custom_fields 2.9.0
Using orm_adapter 0.5.0
Using responders 2.4.0
Using warden 1.2.8
Using devise 4.4.3
Using devise-encryptable 0.2.0
Using dragonfly 1.1.5
Using duktape 1.6.1.0
Using equalizer 0.0.11
Using flash_cookie_session 1.1.6
Using sassc 2.0.0
Using font-awesome-sass 5.0.13
Using mustermann 1.0.3
Using mustermann-grape 1.0.0
Using rack-accept 0.4.5
Using virtus 1.0.5
Using grape 1.1.0
Using grape-entity 0.7.1
Using highline 1.7.10
Using multi_xml 0.6.0
Using httparty 0.16.3
Using jbuilder 2.8.0
Using jquery-rails 4.3.3
Using jquery-ui-rails 6.0.1
Using json-schema 2.8.1
Using kaminari-mongoid 1.0.1
Using kramdown 1.16.2
Using ruby_dep 1.5.0
Using listen 3.1.5
Using locomotivecms-liquid 4.0.0
Using locomotivecms-solid 4.0.1
Using stringex 2.8.5
Using locomotivecms_common 0.3.0
Using mimetype-fu 0.1.2
Using moneta 1.0.0
Using morphine 0.1.1
Using pony 1.12
Using rack-cache 1.7.2
Using rack-rewrite 1.5.1
Using rack_csrf 2.6.0
Using nokogumbo 1.5.0
Using sanitize 4.6.6
Using uglifier 4.1.20
Using locomotivecms_steam 1.4.1
Using mongoid-tree 2.1.1
Using origin 2.3.1
Using pundit 1.1.0
Using rails-i18n 5.1.2
Using simple_form 3.5.1
Using simple_token_authentication 1.15.1
Using temple 0.8.0
Using tilt 2.0.9
Using slim 3.0.9
Using locomotivecms 3.4.0 from https://github.com/locomotivecms/engine.git (at 9ed6083@9ed6083)
Using puma 3.12.0
Using rubyzip 1.2.2
Using sass-rails 5.0.7
Using selenium-webdriver 3.141.0
Using spring 2.0.2
Using spring-watcher-listen 2.0.1
Using turbolinks-source 5.2.0
Using turbolinks 5.2.0
Using web-console 3.7.0
Bundle updated!
~/w/a/locomotiveapp> bundle exec rails server
[DEPRECATION] `use_i18n` is deprecated - use `Money.locale_backend = :currency` instead for currency based formatting
=> Booting Puma
=> Rails 5.1.6.1 application starting in development
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.12.0 (ruby 2.5.3-p105), codename: Llamas in Pajamas
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://0.0.0.0:3000
Use Ctrl-C to stop
Started GET "/locomotive/sign_up" for 127.0.0.1 at 2018-11-30 23:17:02 +0100
[LocomotiveCMS] [fetch site] host = 0.0.0.0 / site_handle = "sign_up" / locale = :en
MONGODB | EVENT: #<Mongo::Monitoring::Event::TopologyOpening topology=Unknown>
MONGODB | Topology type 'unknown' initializing.
MONGODB | EVENT: #<Mongo::Monitoring::Event::ServerOpening address=localhost:27017 topology=Unknown>
MONGODB | Server localhost:27017 initializing.
MONGODB | EVENT: #<Mongo::Monitoring::Event::TopologyChanged prev=Unknown new=Single>
MONGODB | Topology type 'unknown' changed to type 'single'.
MONGODB | EVENT: #<Mongo::Monitoring::Event::ServerDescriptionChanged>
MONGODB | Server description for localhost:27017 changed from 'unknown' to 'standalone'.
MONGODB | EVENT: #<Mongo::Monitoring::Event::TopologyChanged prev=Single new=Single>
MONGODB | There was a change in the members of the 'single' topology.
MONGODB | localhost:27017 | locomotiveapp_development.find | STARTED | {"find"=>"locomotive_sites", "filter"=>{"handle"=>"sign_up"}, "sort"=>{"_id"=>1}, "limit"=>1, "singleBatch"=>true}
MONGODB | localhost:27017 | locomotiveapp_development.find | SUCCEEDED | 0.001s
Processing by Locomotive::RegistrationsController#new as HTML
  Rendering /Users/gnoireaux/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/bundler/gems/engine-9ed6083539b5/app/views/locomotive/registrations/new.html.slim within locomotive/layouts/account
  Rendered /Users/gnoireaux/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/bundler/gems/engine-9ed6083539b5/app/views/locomotive/registrations/new.html.slim within locomotive/layouts/account (57532.5ms)
Completed 200 OK in 59130ms (Views: 59118.0ms)
```

**Comment by DonKoko** [comment]
Issue still persists, fix still works.

**Comment by did** [comment]
sorry people, fixed here #cae3f9abcd16c1a3b85034228f281b6aee0c847c. 
btw, I've released a 3.4.1 version.
